RocketChat/Rocket.Chat
# feat: [ENTERPRISE] Add setting to control user merge on LDAP Background Sync

## Proposed changes (including videos or screenshots)


 - Allow workspaces to merge existing users' LDAP info on background sync (even when the "Background Sync Import New Users" setting is disabled);
 - Add "Background Sync Merge Existing Users" setting under **LDAP > Enterprise > Background Sync** settings (this setting can only be enabled when "Update Existing Users" is also enabled).



## Issue(s)


## Steps to test or reproduce

Go to **Workspace > Settings > LDAP > Enterprise**, disable "Background Sync Import New Users" and enable "Background Sync Update Existing Users" setting. Login for the first time with an user via OAuth and wait for a background sync to happen.
**_Current behavior:_** the user that logged in via OAuth won't have their LDAP data synced on background sync.
**_Expected behavior:_** after enabling the new "Background Sync Merge Existing Users" setting, the user that logged in via OAuth should have their LDAP data merged on background sync.

## Further comments

TC-433
